### PR TITLE
Add service registration test

### PR DESCRIPTION
**Pull request by B3rn475**

### CODE REVIEW

File: dart/example/vm_service_lib_tester.dart [hunk 1]
```diff
@@ -62,13 +62,42 @@ main(List<String> args) async {
...
+      (Map<String, dynamic> params) async {
+    return <String, dynamic>{'output': params['input']};
+  });
+  await serviceClient.registerService(serviceName, serviceAlias);
+  VmService otherClient =
+      await vmServiceConnect(host, port, log: new StdoutLog());
+  Completer completer = new Completer();
+  otherClient.onServiceEvent.listen((e) async {
+    if (e.service == serviceName) {
+      assert(e.alias == serviceAlias);
```
Comment (devoncarew) [inline dart/example/vm_service_lib_tester.dart]: It looks like this assert is failing on travis.

File: java/src/org/dartlang/vm/service/RemoteServiceRunner.java [hunk 2]
```diff
@@ -21,9 +21,8 @@
 public interface RemoteServiceRunner {
```
Comment (devoncarew) [inline java/src/org/dartlang/vm/service/RemoteServiceRunner.java]: You could also have `RemoteServiceRunner` have a method like `getServiceName()`, so you don't need to pass the serviceName into the addServiceRunner method - a RemoteServiceRunner would know both its name and how to handle service call.s

**Comment by devoncarew** [comment]
https://travis-ci.org/dart-lang/vm_service_drivers/jobs/260406646#L301

File: dart/example/vm_service_lib_tester.dart (hunk 1 above)
Comment (B3rn475) [inline dart/example/vm_service_lib_tester.dart]: Done

File: java/src/org/dartlang/vm/service/RemoteServiceRunner.java (hunk 2 above)
Comment (B3rn475) [inline java/src/org/dartlang/vm/service/RemoteServiceRunner.java]: Passing the `serviceName` into `assServiceRunner` inforces the static nature of the value.
A getter can potentially return different values.
Another reason is to keep the same approach used for the Completer interface.

Comment (devoncarew) [inline java/src/org/dartlang/vm/service/RemoteServiceRunner.java]: gotcha

**Comment by devoncarew** [comment]
lgtm!

**Pull request merged**